### PR TITLE
[JW8-11075] Add polyfill for scroll options and prevent auto scrolls

### PR DESCRIPTION
**Pull request by zetagame**
### This PR will...
Polyfill the prevent scroll option and use it for any auto focus events.

### Why is this Pull Request needed?
It is jarring when hiding controls focuses the player scrolling it into view.

### Are there any Pull Requests open in other repos which need to be merged with this?
https://github.com/jwplayer/jwplayer-commercial/pull/7625

#### Addresses Issue(s):

JW8-11075

### Checklist
- [ ] Jenkins builds and unit tests are passing
- [ ] I have reviewed the automated results

### CODE REVIEW

File: src/js/jwplayer.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import 'polyfills/promise';
+import 'polyfills/focus';
```
Comment (robwalch) [inline src/js/jwplayer.js]: Please import this in controls. It's not needed outside of controls, and would exclude it from the headless player without special logic.

File: src/js/polyfills/focus.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+/* eslint getter-return: 0, no-empty: 0 */
```
Comment (robwalch) [inline src/js/polyfills/focus.js:1]: Did this code come from https://github.com/calvellido/focus-options-polyfill/blob/master/index.js
?

Open-source added to the player needs to be attributed with accordance to it's license. (https://github.com/calvellido/focus-options-polyfill/blob/master/LICENSE)

You'll need to make the appropriate changes in:
- [ ] https://github.com/jwplayer/jwplayer/blob/master/jwplayer.license.notice.js
- [ ] https://github.com/jwplayer/jwplayer-commercial/blob/master/jwplayer.license.notice.js
to include this project in the notice.txt that is output with each build.

Comment (zetagame) [reply src/js/polyfills/focus.js:1]: Thanks!

Comment (robwalch) [reply src/js/polyfills/focus.js:1]: If this did not require any modification then can we bring it via yarn/npm?

File: jwplayer.license.notice.js [hunk 3]
```diff
@@ -4,6 +4,7 @@ const YEAR = new Date().getFullYear();
 const versions = {
     backbone: '1.1.2',
     'can-autoplay': '3.0.0',
+    'focus-options-polyfill': '1.5.0',
```
Comment (robwalch) [inline jwplayer.license.notice.js:7]: If this package was imported via yarn/npm then we could do
```suggestion
    'focus-options-polyfill': devDeps['focus-options-polyfill'],
```

**Comment by zetagame** [comment]
test this please

**Comment by zetagame** [comment]
test this please

File: src/js/view/controls/controls.js [hunk 4]
```diff
@@ -19,6 +19,8 @@ import ShortcutsTooltip from 'view/controls/shortcuts-tooltip';
 import FloatingCloseButton from 'view/floating-close-button';
 
 require('css/controls.less');
+// Allows preventScrolling focus option on IE/Safari.
+require('../../../../node_modules/focus-options-polyfill');
```
Comment (jnatalzia) [inline src/js/view/controls/controls.js:23]: we should put an import into `src/js/polyfills` and import that here for clarity and file structure. that file in `src/js/polyfills` can simply run this require

**Comment by zetagame** [comment]
test this please

**Comment by zetagame** [comment]
test this please

**Pull request merged**